simpleviewinc/keg-cli
# fix(env): Fix initial load of container

### Context
* This fix is for **internal** only apps managed by the Keg-CLI *( keg-core | keg-components | keg-proxy )*
* At the time that their container meta data is loaded, the default env is unknown
* Because of this, it's not possible to use override files, to override the default envs
* Typically an override files would look like `~/.kegConfig/proxy-staging.env`
  *  This would override the proxy envs when in a staging environment
  * But because the env is unknown at the time these files are loaded
  * The `proxy-staging.env` never gets loaded, and does not override the default

### Goal
* Ensure the current env is known before the containers are loaded for internal apps
* Remove code that's been extracted to other packages, or no longer used

### Updates

**Env Fix**
* `src/constants/constants.js`
  * Extract `ENV_MAP` out of main constants object so it can be reused
  * Added `ENV_OPTIONS` constant, which is an array of all allowed environments
* `src/constants/docker/containers.js`
  * Added code to get the env before loading the internal containers
  * This ensure any overwrite files can be loaded properly
  * This ensure the config files based on env get loaded
  * Previously this only worked for injected apps
* `src/utils/helpers/checkArgsForEnv.js` 
  * Added helper to load the correct env when loading containers
  * The internal apps containers get loaded prior to the command line arguments being parsed
    * Which means we don't have access to the args.param object like tasks do
  * This helper searches the command line arguments for an `environment` argument, or an alias
    * If found, it is returned, otherwise it returns `false`
  * It also maps the env shortcuts to the real value, i.e. `prod` => `production` || `dev` => `development`

**Code Cleanup**
* Updated all references to the local `exists` helper to use the `jsutils` version
  * This was added before `jsutils` had it, and it's no longer needed 
* `src/utils/helpers/mapEnv.js`
  * This code is not used in the keg-cli any longer, so I removed it 
* `src/utils/helpers/limboify.js`
  * Not used anywhere in the keg-cli
  * The `FileSysLib` has it's own version of it, defined with the module 

### Tests
**Test 1**
Pull this pr => `keg cli && keg pr 38`
Run the tests `yarn test`
  * Ensure all tests pass

**Test 2**
* Stop the keg-proxy if it's running =>  `keg proxy kill` 
* Create and override file at `~/.kegConfig/proxy-staging.env`
  * Override the `KEG_PROXY_HOST` env with `some.other.url`
    * Looks like this =>  http://snpy.in/hOenJU
* Restart the proxy with `keg proxy start --env st`
* Once started, run the command `docker container inspect keg-proxy`
  * In the inspect output, search for `staging.kegdev.xyz`
  * You should find a label on the container with that value set
  * Looks like this => http://snpy.in/Tj1kc0
